### PR TITLE
Bug fixes while testing with Rhapsody files

### PR DESCRIPTION
**Pull request by raddanki-eq**
Bug fixes while testing with Rhapsody files

### CODE REVIEW

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java [hunk 1]
```diff
@@ -171,26 +169,11 @@ private Hl7ToRhapsodysXmlConverter() {
...
         String rhapsodyXml = "";
 
         HL7Helper hl7Helper = new HL7Helper();
-
-
         HL7ParsedMessage hl7ParsedMsg = hl7Helper.hl7StringParser(hl7Msg);
 
-        // this should be handled by validation pipeline
-        // the one scenario this might be needed is when someone reprocessing the failed message on this pipeline
-        switch (hl7ParsedMsg.getOriginalVersion()) {
```
Comment (ndduc01) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:184]: could you leave this as it is for now, also ignore the comment above the code. 
This basically convert 231 to 251 if applicable.

I will move this into library function

Comment (raddanki-eq) [reply report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:184]: Will fix in next release after api's get updated.

**Comment by sdeusch** [comment]
**Dead Code**

**Readability, Compactness **
Instead of 
       if (null != ce.getAlternateIdentifier()) && (ce.getAlternateIdentifier().length() > 0) 
you could leverage utility APIs, e.g., 
       if( StringUtils.isEmpty(ce.getAlternateIdentifier()) 

** Unit test for the long methods? **

**Comment by raddanki-eq** [comment]
Hi Stefan, as discussed on slack, I like StingUtil's suggestion, but, that jar is not imported, thus, I depended on native methods, I will change that as part of sprint 25, in a separate PR. Pls go ahead merge if that's the only issue.

**Comment by raddanki-eq** [comment]
> **Dead Code**
> 
> **Readability, Compactness** Instead of if (null != ce.getAlternateIdentifier()) && (ce.getAlternateIdentifier().length() > 0) you could leverage utility APIs, e.g., if( StringUtils.isEmpty(ce.getAlternateIdentifier())
> 
> **Unit test for the long methods?**

Hi Stefan, not sure we are including apache string util's lib, I will make suggested change as part of sprint-25. Pls merge pending work, will take that up as next story.

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java [hunk 2]
```diff
@@ -206,16 +193,41 @@ public String convert(String hl7Msg) throws Exception {
...
+
         return rhapsodyXml;
     }
 
+    private void saveXmlToTempFile(String rhapsodyXml) {
+        try {
+            java.nio.file.Path path = java.nio.file.Paths.get("/Users/RameshAddanki/downloads/gen101.xml");
+            java.nio.file.Files.writeString(path, rhapsodyXml, java.nio.charset.StandardCharsets.UTF_8);
+        } catch (java.io.IOException ex) {
+            System.out.print("Invalid Path");
```
Comment (bspuli) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:212]: Catch this exception properly.

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java [hunk 3]
```diff
@@ -1,6 +1,5 @@
 package gov.cdc.dataingestion.nbs.converters;
```
Comment (bspuli) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:2]: This file is too big and lots of if and else conditions. Is there any better way to make it simple

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java [hunk 4]
```diff
@@ -393,14 +413,38 @@ private HL7CNEType buildHL7CNEType(Cne cne) {
...
-        hl7CNEType.setHL7Text(cne.getText());
-        hl7CNEType.setHL7NameofCodingSystem(cne.getNameOfAlternateCodingSystem());
-        hl7CNEType.setHL7AlternateIdentifier(cne.getAlternateIdentifier());
-        hl7CNEType.setHL7AlternateText(cne.getAlternateText());
-        hl7CNEType.setHL7NameofAlternateCodingSystem(cne.getNameOfAlternateCodingSystem());
-        hl7CNEType.setHL7CodingSystemVersionID(cne.getCodingSystemVersionId());
-        hl7CNEType.setHL7AlternateCodingSystemVersionID(cne.getAlternateCodingSystemVersionId());
-        hl7CNEType.setHL7OriginalText(cne.getOriginalText());
+
+        if ((null != cne.getText()) && (cne.getText().length() > 0)) {
```
Comment (bspuli) [inline report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:417]: may be we can use common util function do null check

Comment (raddanki-eq) [reply report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:417]: Stefan suggested the same. Pls see my notes on his comments. Tx

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java (hunk 2 above)
Comment (raddanki-eq) [reply report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java:212]: Done

File: report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java (hunk 4 above)
Comment (raddanki-eq) [reply report-service/src/main/java/gov/cdc/dataingestion/nbs/converters/Hl7ToRhapsodysXmlConverter.java]: Done

**Comment by sdeusch** [comment]
Still a lot of dead code, see https://sonarcloud.io/project/issues?resolved=false&severities=MAJOR&types=CODE_SMELL&pullRequest=49&id=CDCgov_NEDSS-DataIngestion 

Unit tests?

**Review by sdeusch — APPROVED** [review]

**Pull request merged**